iceoryx_posh/source/runtime/service_discovery.cpp

```diff
@@ -39,11 +42,16 @@ ServiceDiscovery::findService(const cxx::optional<capro::IdString_t>& service,
     {
         instanceString = instance.value();
     }
+    if (!isEventWildcard)
```

Comment:
Can probably now use `if(event)` and get rid of the helper `bool isEventWildcard` (introduced before `optional`) and similar `bool` values. Will also get rid of redundat negation etc.

Reply:
Correct, but I kept `is*Wildcard` booleans to make the intention clearer in line 52ff. But no hard feelings about this.

Reply:
I would prefer to get rid of them because for me `if(event)` reads nicely as "if the event exists" (i.e. is no wildcard). Gets rid of unnecessary definitions and saves (maybe optimized) negations.

Reply:
3:1 for keeping `is*Wildcard`.